src/util/kaldi-io.cc

```diff
@@ -34,16 +34,27 @@
 #define MapOsPath(x) x
 #endif  // KALDI_CYGWIN_COMPAT
 
-#ifdef _MSC_VER
+
+#if defined(_MSC_VER) || defined(__CYGWIN__)
 static FILE *popen(const char* command, const char* mode) {
 #ifdef KALDI_CYGWIN_COMPAT
   return kaldi::CygwinCompatPopen(command, mode);
+#elif defined(__CYGWIN__)
+  return popen(command, mode);
 #else  // KALDI_CYGWIN_COMPAT
   return _popen(command, mode);
 #endif  // KALDI_CYGWIN_COMPAT
 }
 #endif  // _MSC_VER
 
+#ifdef __CYGWIN__
+static pclose(FILE* file) {
+  //FILE *file = (FILE*)_file;
```

Comment:
I don't see the purpose of this function -- seems almost like a recursive call or somethign
y.

Reply:
Sorry, this change(about 'popen' and 'pclose') is a mistake. The original code was right. I find that what I need to change is in the /usr/include/stdio.h rather than in kaldi.